File: web/src/types/console.ts
import type { LogListResponse } from './logs'
import type { Node, ProtocolOption } from './node'
import type { Subscription } from './subscription'
import type { TunStatusState } from './tun'

export interface RefreshContext {
  refreshConsoleAndNodes: () => Promise<void>
  refreshLogsSilently: () => Promise<void>
  showLogs: () => void
}

export interface ConsoleSnapshot {
  node_summary: {
    all: number
    available: number
    pending: number
    timeout: number
  }
  runtime: {
    running: boolean
    tun_enabled: boolean
    tun_state: TunStatusState
    proxy: {
      enabled: boolean
      http_port: number
      socks_port: number
    }
    current_node?: Node
  }
  subscriptions: Subscription[]
  protocols: ProtocolOption[]
  logs: LogListResponse
}

File: server/internal/dto/console.go
package dto

import "v2ray-server/internal/service"

type ConsoleSnapshot struct {
	NodeSummary   service.NodeSummary      `json:"node_summary"`
	Runtime       ConsoleRuntimeDTO        `json:"runtime"`
	Subscriptions []SubscriptionDTO        `json:"subscriptions"`
	Protocols     []service.ProtocolOption `json:"protocols"`
	Logs          ConsoleLogsDTO           `json:"logs"`
}

type ConsoleRuntimeDTO struct {
	Running     bool              `json:"running"`
	TunEnabled  bool              `json:"tun_enabled"`
	TunState    string            `json:"tun_state"`
	Proxy       ProxyDTO          `json:"proxy"`
	CurrentNode *service.NodeInfo `json:"current_node,omitempty"`
}

type ProxyDTO struct {
	Enabled   bool `json:"enabled"`
	HTTPPort  int  `json:"http_port"`
	SOCKSPort int  `json:"socks_port"`
}

type ConsoleLogsDTO struct {
	Items      []LogDTO `json:"items"`
	Tags       []string `json:"tags"`
	Levels     []string `json:"levels"`
	NextCursor string   `json:"next_cursor"`
	HasMore    bool     `json:"has_more"`
}
